esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -2848,29 +2846,30 @@ bool Cws_machineEx::onGetComponentUsage(IEspContext& context, IEspGetComponentUs
 
         StringBuffer timeStr;
         IArrayOf<IEspComponentUsage> componentUsages;
-        Owned<IPropertyTree> usageReq = createPTree("Req");
+        Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
+        Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
+        Owned<IPropertyTree> usageReq = readComponentUsageReq(req, constEnv);
 
         Owned<CUsageCache> usage;
         Owned<IPropertyTree> uniqueUsages;
-        if (!req.getBypassCachedResult())
+        if (req.getBypassCachedResult())
         {
-            usage.setown((CUsageCache*) usageCacheReader->getCachedInfo());
-            if (!usage)
-                throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Failed to get usage. Please try later.");
+            uniqueUsages.setown(createPTree("Usage"));
+            //Add unique machines from the usageReq to uniqueUsages.
+            setUniqueMachineUsageReq(usageReq, uniqueUsages);
+            getMachineUsages(context, uniqueUsages);
         }
         else
         {
-            uniqueUsages.setown(createPTree("Usage"));
-        }
+            usage.setown((CUsageCache*) usageCacheReader->getCachedInfo());
+            if (!usage)
+                throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Failed to get usage. Please try later.");
 
-        Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
-        Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
-        readComponentUsageReq(req, constEnv, usageReq, uniqueUsages);
-        if (req.getBypassCachedResult())
-        {
-            getMachineUsages(context, uniqueUsages);
+            ESPLOG(LogMax, "GetComponentUsage: found UsageCache.");
+            uniqueUsages.set((IPropertyTree*) usage->queryUsages());
```

Comment:
not good practice to remove const via cast like this.
Would be better to make 'uniqueUsages' a Owned<const IPropertyTree>,
and at line 2857 do:
```
            Owned<IPropertyTree> tmpUniqueUsages = createPTree("Usage");
            //Add unique machines from the usageReq to uniqueUsages.
            setUniqueMachineUsageReq(usageReq, tmpUniqueUsages);
            uniqueUsages.setown(tmpUniqueUsages.getClear());
```